npm/src/filechecks/get-file-check-progress.ts

```diff
@@ -34,33 +33,31 @@ export const getFileChecksInfo: (
   try {
     const result: FetchResult<GetFileCheckProgressQuery> =
       await client.mutation(GetFileCheckProgress, variables)
+
     if (!result.data || result.errors) {
       const errorMessage: string = result.errors
         ? result.errors.toString()
         : "no data"
-      throw Error("Add files failed: " + errorMessage)
+      throw Error(`Add files failed: ${errorMessage}`)
     } else {
       const getConsignment = result.data.getConsignment
+
       if (getConsignment) {
         const fileChecks = getConsignment.fileChecks
-        const totalFiles = getConsignment.totalFiles
-        const antivirusProcessed = fileChecks.antivirusProgress.filesProcessed
-        const checksumProcessed = fileChecks.checksumProgress.filesProcessed
-        const ffidProcessed = fileChecks.ffidProgress.filesProcessed
-        return callback({
-          antivirusProcessed,
-          checksumProcessed,
-          ffidProcessed,
-          totalFiles
-        })
+        return {
+          antivirusProcessed: fileChecks.antivirusProgress.filesProcessed,
+          checksumProcessed: fileChecks.checksumProgress.filesProcessed,
+          ffidProcessed: fileChecks.ffidProgress.filesProcessed,
+          totalFiles: getConsignment.totalFiles
+        }
       } else {
         console.log(
           `No progress metadata found for consignment ${consignmentId}`
```

Comment:
Does this case ever actually happen? i.e. there are no errors in the GraphQL response, but nothing in the `data` response either. I wonder if it would be better to throw an error instead? Then you could get simplify the `IFileCheckProgress | null` return type.

Reply:
You're referring to Line 58 (in the old version, Line 55 in the new)? I'm not sure, Sam added this. @MancunianSam do you know of a case?

@suzannehamilton but even if we don't return null here, the `catch` further down does (in case the response fails). Also, I don't know how hard it would make it to test for exceptions; I tried but I couldn't get it to work, but I could have been doing something wrong.

Reply:
Do you want to have a call about the exception handling to see if we can get something to work?

Reply:
Sure!